x-pack/plugins/security_solution/public/management/cypress/e2e/artifacts.cy.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import {
+  ENDPOINT_ARTIFACT_LIST_IDS,
+  EXCEPTION_LIST_URL,
+} from '@kbn/securitysolution-list-constants';
+import { BASE_ENDPOINT_ROUTE } from '../../../../common/endpoint/constants';
+import { login, loginWithRole, ROLE } from '../tasks/login';
+
+import { type FormAction, getArtifactsListTestsData } from './mocks/artifacts_page';
+import { runEndpointLoaderScript } from '../tasks/run_endpoint_loader';
+
+const removeAllArtifacts = () => {
+  for (const listId of ENDPOINT_ARTIFACT_LIST_IDS) {
+    cy.request({
+      method: 'DELETE',
+      url: `${EXCEPTION_LIST_URL}?list_id=${listId}&namespace_type=agnostic`,
+      headers: { 'kbn-xsrf': 'kibana' },
+      failOnStatusCode: false,
+    });
+  }
+};
+
+const loginWithWriteAccess = (url: string) => {
+  loginWithRole(ROLE.analyst_hunter);
+  cy.visit(url);
+};
+
+const loginWithReadAccess = (url: string) => {
+  loginWithRole(ROLE.t2_analyst);
+  cy.visit(url);
+};
+
+const loginWithoutAccess = (url: string) => {
+  loginWithRole(ROLE.t1_analyst);
+  cy.visit(url);
+};
+
+// Checks for Endpoint data and creates it if needed
+const loadEndpointDataForEventFiltersIfNeeded = () => {
+  cy.request({
+    method: 'POST',
+    url: `${BASE_ENDPOINT_ROUTE}/suggestions/eventFilters`,
+    body: {
+      field: 'agent.type',
+      query: '',
+    },
+    headers: { 'kbn-xsrf': 'kibana' },
+    failOnStatusCode: false,
+  }).then(({ body }) => {
+    if (isEmpty(body)) {
+      runEndpointLoaderScript();
+    }
+  });
+};
+
+const runAction = (action: FormAction) => {
+  if (action.type === 'click') {
+    cy.getBySel(action.selector).click();
+  } else if (action.type === 'input') {
+    cy.getBySel(action.selector).type(action.value || '');
+  } else if (action.type === 'clear') {
+    cy.getBySel(action.selector).clear();
+  } else if (action.type === 'customClick') {
+    cy.get(action.selector).click();
+  } else if (action.type === 'removeItem') {
+    cy.get(`[data-test-subj="${action.selector}"] > span > button`).click();
+  }
+};
+
+describe('Artifacts pages', () => {
+  before(() => {
+    login();
+    loadEndpointDataForEventFiltersIfNeeded();
+    // Clean artifacts data
+    removeAllArtifacts();
```

Comment:
Why do a clean in the `before`?

Reply:
Because we want to remove existing data (from other test or other executions) before start running all test cases, for example, in order to test the empty pages for artifact lists

Reply:
👍 

_(🤔  but does that mean that we have tests that are leaving data around?)_

Reply:
If other test has an unexpected termination the data could be there I guess, it is just a way to ensure a clean stage.